Utils/PersistentDataStore.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.IO;
+using UnityEngine;
+
+namespace DUCK.Utils
+{
+	public static class PersistentDataStore
+	{
+		private static readonly string SaveDataPath = Application.persistentDataPath + "/PersistentData/";
+
+		public static bool Exists<T>(string uid = "")
+		{
+			return File.Exists(GetFilePath<T>(uid));
+		}
+
+		public static void Save<T>(T dataObject, string uid = "")
+		{
+			if (dataObject == null) throw new ArgumentException("Cannot save a null object.");
+
+			File.WriteAllText(GetFilePath<T>(uid), JsonUtility.ToJson(dataObject));
+		}
+
+		public static T Load<T>(string uid = "")
+		{
+			var path = GetFilePath<T>(uid);
+
+			try
+			{
+				return JsonUtility.FromJson<T>(File.ReadAllText(path));
+			}
+			catch
+			{
+				return default(T);
```

Comment:
this will now return null then unless it's a struct

Reply:
Yes - we have the Exists<> function to explicitly check if data has been stored, or the caller can just null/empty-check the result.